gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/Misc.java

```diff
@@ -169,6 +170,18 @@ public final static GemFireStore getMemStoreBootingNoThrow() {
     return GemFireStore.getBootingInstance();
   }
 
+  public static void waitForSamplerInitialization() {
+    final GemFireStatSampler sampler = getDistributedSystem().getStatSampler();
+    if (sampler != null) {
+      try {
+        sampler.waitForInitialization();
```

Comment:
Would you want to use the timed wait version instead? and fail if for whatever reason that does not get initialized.

Reply:
Ok, will make it timeout after ackWaitThreshold (which is 15s by default after which normal messages spit out "15 seconds elapsed...").